File: frontend/src/types/order.ts
export interface Order {
  id: string;
  ebay_order_id: string;
  buyer_username: string;
  order_status: string;
  payment_status: string;
  total_amount: number;
  subtotal: number;
  shipping_cost: number;
  tax_amount: number;
  ebay_fees_total: number;
  shipping_carrier?: string;
  tracking_number?: string;
  shipped_at?: string;
  paid_at?: string;
  aliexpress_order_id?: string;
  aliexpress_order_status?: string;
  aliexpress_tracking_number?: string;
  shipping_address: ShippingAddress;
  line_items: OrderLineItem[];
  created_at: string;
  updated_at: string;
}

export interface ShippingAddress {
  name: string;
  addressLine1: string;
  addressLine2?: string;
  city: string;
  stateOrProvince: string;
  postalCode: string;
  countryCode: string;
}

export interface OrderLineItem {
  id: string;
  ebay_line_item_id: string;
  title: string;
  sku: string;
  quantity: number;
  unit_price: number;
  total_price: number;
  ebay_final_value_fee?: number;
  ebay_payment_processing_fee?: number;
  ebay_promoted_listing_fee?: number;
  listing_id?: string;
  product_id?: string;
}

export interface OrderListItem {
  id: string;
  ebay_order_id: string;
  buyer_username: string;
  order_status: string;
  payment_status: string;
  total_amount: number;
  tracking_number?: string;
  paid_at?: string;
  created_at: string;
}

export type OrderStatus =
  | "NOT_STARTED"
  | "IN_PROGRESS"
  | "FULFILLED"
  | "CANCELLED";

export type PaymentStatus = "PENDING" | "PAID" | "FAILED" | "REFUNDED";

File: backend/app/schemas/order.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class OrderLineItemResponse(BaseModel):
    id: uuid.UUID
    ebay_line_item_id: str | None = None
    title: str
    sku: str | None = None
    quantity: int
    unit_price: float
    total_price: float
    ebay_final_value_fee: float | None = None
    ebay_payment_processing_fee: float | None = None
    ebay_promoted_listing_fee: float | None = None
    ebay_international_fee: float | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OrderResponse(BaseModel):
    id: uuid.UUID
    marketplace: str
    ebay_order_id: str
    buyer_username: str | None = None
    order_status: str
    payment_status: str
    total_amount: float
    subtotal: float
    shipping_cost: float
    tax_amount: float
    ebay_fees_total: float | None = None
    currency: str
    shipping_address: dict | None = None
    shipping_carrier: str | None = None
    tracking_number: str | None = None
    shipped_at: datetime | None = None
    paid_at: datetime | None = None
    aliexpress_order_id: str | None = None
    aliexpress_order_status: str | None = None
    aliexpress_tracking_number: str | None = None
    line_items: list[OrderLineItemResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OrderShipRequest(BaseModel):
    shipping_carrier: str
    tracking_number: str


class OrderListResponse(BaseModel):
    id: uuid.UUID
    ebay_order_id: str
    buyer_username: str | None = None
    order_status: str
    payment_status: str
    total_amount: float
    currency: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
